Adding QConfigTypePtrMap

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [WIP][quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32380 [quant][graphmode][refactor] Better API for fold_convbn
* #32374 [refactor] Adding FoldConvBatchNorm2dHelper
* #32303 [quant][graphmode] Call _jit_pass_dedup_module_ueses in quantize_script
* #32302 [quant] Re-enable fold_convbn in quantize_script
* #32206 [quant] Re-enable test_nested that has different qconfig for shared ClassType
* #32205 [quant][graphmode] Support quantizing shared ClassType with different qconfigs
* #32204 [quant][graphmode] Default to non-inplace in graph mode quantization API
* **#32203 Adding QConfigTypePtrMap**
* #32202 [refactor] Move some of the helper functions for public use
* #31408 [quant][graphmode] `insert_quant_dequant` pass support shared class types

Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508027](https://our.internmc.facebook.com/intern/diff/D19508027)

jerryzh168 committed:
Adding QConfigTypePtrMap

Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit a1070006:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=a1070006e6753f1a5cf17fb67b001bdcb8ee73cf).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4248281) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=86222988) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=86222988))
<details>
<summary>
<code>Jan 15 02:41:13 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 15 02:41:13 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 15 02:41:13 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:41:13 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 15 02:41:13 1 out of 1 hunk FAILED 
Jan 15 02:41:13 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:41:13 Traceback (most recent call last): 
Jan 15 02:41:13   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 15 02:41:13     patch_repo(args) 
Jan 15 02:41:13   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 15 02:41:13     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 15 02:41:13 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 15 02:41:13 + cleanup 
Jan 15 02:41:13 + retcode=1 
Jan 15 02:41:13 + set +x 
Jan 15 02:41:13 =================== sccache compilation log =================== 
Jan 15 02:41:13 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 15 02:41:13 Compile requests                 1 
Jan 15 02:41:13 Compile requests executed        0 
Jan 15 02:41:13 Cache hits                       0 
Jan 15 02:41:13 Cache misses                     0 
Jan 15 02:41:13 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good, but please run clang-format on the entire stack before landing.

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "Adding QConfigTypePtrMap"


Summary:
The type is needed for allowing multiple qconfig configurations for shared
ClassType, see next PR for more details

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
